### PR TITLE
added clean for BNodes

### PR DESCRIPTION
**Pull request by cedricdcc**
After the current changes, the `reparse` workaround will now **behave successfully and correctly**, without any crashes. 

Here is the exact breakdown of the before and after behavior:

### Before the Current Changes (Why it crashed)
1. **JSON-LD Parse:** The JSON-LD parser reads `_:help@embrc.org` and creates `BNode("help@embrc.org")`.
2. **Incorrect Cleaner Order:** In the unpatched `build_clean_chain`, the graph-level cleaners (`grouped_fn[Level.Graph]`) executed in definition order. Since [reparse](file:///c:/Users/cedricd/Documents/Github/py-sema/sema/commons/clean/clean.py#L22-L32) is a graph-level cleaner, and the node-level cleaners were *appended* to the list, `reparse` ran **first** on the raw graph.
3. **Serialization & Crash:** Inside `reparse`, `g.serialize(format="nt")` output the string `_:help@embrc.org ...`. When `Graph().parse(..., format="nt")` attempted to parse this back, it crashed because `@` is an invalid character in the W3C N-Triples blank node grammar.

---

### After the Current Changes
1. **JSON-LD Parse:** The JSON-LD parser reads `_:help@embrc.org` and creates `BNode("help@embrc.org")`.
2. **Corrected Cleaner Order:** The node and triple level cleaners (including the new `clean_bnode_node`) are now **prepended** at index `0` of the graph-level cleaner chain.
3. **Sanitization:** The node-level cleaner `clean_bnode_node` executes **first**. It scans the graph nodes, catches `BNode("help@embrc.org")`, replaces the invalid `@` with `_`, and yields `BNode("help_embrc.org")`.
4. **Reparsing Succeeded:** Next, `reparse` executes on the sanitized graph:
   * `g.serialize(format="nt")` outputs `_:help_embrc.org ...` which is valid N-Triples syntax.
   * `Graph().parse(..., format="nt")` parses it successfully.
   * Because of this parse-serialize-parse cycle, the local BNode `BNode("help_embrc.org")` is mapped to a **new, fresh blank node** (e.g. `BNode("N6bb34be2a5cd4be6a964df876137413d")`), successfully resolving the original rdflib issue #2720.
  

## Summary by CodeRabbit

* **Bug Fixes**
  * Changed the default intermediate round-tripping format for graph cleaning/parsing to Turtle.
  * Updated the cleaning pipeline order so triple-level cleaning runs before later graph-level steps for more consistent results.
* **Tests**
  * Added parametrized coverage for JSON-LD blank-node and non-blank IRI identifiers, including special-character edge cases, across multiple RDF formats; verifies parsing, cleaning, triple counts, and Turtle serialization.
* **Chores**
  * Updated ignore rules to prevent `.agents` and `skills-lock.json` from being tracked by git.

### CODE REVIEW

File: tests/commons/clean/test_blank_nodes.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        with pytest.raises(Exception):
+            clean_graph(g, cleaner)
+    else:
+        cleaned_g = clean_graph(g, cleaner)
+        assert len(cleaned_g) == len(g)
+        # Verify serialization completes without raising
+        cleaned_g.serialize(format="ttl")
+
+
+def test_other_rdf_formats_invalid_blank_node_failures():
```
Comment (mpo-vliz) [inline tests/commons/clean/test_blank_nodes.py]: this test does not add any value -- it only parses never cleans -- so it is about rdflib at best, never about py-sema -- so actually also the json-ld case would pass here

**Review by mpo-vliz — CHANGES_REQUESTED** [review]
please reconsider the testcase
